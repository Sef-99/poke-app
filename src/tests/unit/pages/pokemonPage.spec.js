import { vi, it, describe, beforeEach, expect } from 'vitest';
import { mount, shallowMount } from '@vue/test-utils';
import PokemonPage from '@/pages/PokemonPage.vue';
import { pokemonArrayMock } from '@/tests/unit/mocks/pokemonArray.mock.js';

describe('Pokemon page related tests', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(PokemonPage);
  });

  it('should match with snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should match snapshot with pokemons', () => {
    const wrapper = shallowMount(PokemonPage, {
      data() {
        return {
          pokemonOptionsList: pokemonArrayMock,
        };
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
