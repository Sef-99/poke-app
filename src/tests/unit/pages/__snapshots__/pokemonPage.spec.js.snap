// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pokemon page related tests > should match snapshot with pokemons 1`] = `
"<div class="progress-container">
  <v-progress-circular indeterminate=""></v-progress-circular>
</div>"
`;

exports[`Pokemon page related tests > should match with snapshot 1`] = `
"<div class="progress-container">
  <v-progress-circular indeterminate=""></v-progress-circular>
</div>"
`;
